docs/source/dev/yaml_spec.rst

```diff
@@ -0,0 +1,425 @@
+YAML Data File Specification
+============================
+
+This file constitutes the official specification for the YAML data files used
+to store the data for an :term:`instrument`.
+
+Spec
+----
+
+The data must be stored in a YAML format file (indentation matters!) with the
+following structure:
+
+.. parsed-code-block:: yaml
```

Comment:
Check out https://plantuml.com/yaml

Reply:
I am not sure that this is a place for a diagram? I think we want to show the YAML file as-is here for reference, and `parsed-code-block` is used to be able to have cross-links between the YAML file lines and the text below. What do you think, @ajjackson ?

Reply:
I think it can sometimes be quite handy to have a visual representation, but it's just something to consider rather than a this must happen.

Reply:
I don't find those diagrams any more legible than the yaml source

Reply:
Fair enough, I find that breaking out what is e.g. a valid version spec into its own chunk highlights it a bit. Particularly if you have optional/mandatory bits you can highlight them with bolding and things. 

Happy for this to be marked as resolved, as noted, just something to consider.

Reply:
Could be useful in the how-to section?